test/testDB.js

```diff
@@ -0,0 +1,133 @@
+const { avatarUploaderMiddleware } = require('../middlewares/fileUploader.middleware');
+const { uploadAvatarController } = require('../api/auth/auth.controller');
+
+describe('Connection mongo', () => {
+  it("Connection mongo test with valid URI", (done) => {
+    connect(process.env.DB_URI, { useNewUrlParser: true, useUnifiedTopology: true }, (err) => {
+      if (err) done(err);
+      done();
+    });
+  });
+})
+
+describe('Unit test for auth middleware', () => {
+  const argForMiddleware = () => {
+    const req = mockReq();
+    const res = mockRes({ params: {} });
+    const next = sinon.stub();
+  
+    res.status = function (params) {
+      this.params.status = params;
+      return this;
+    };
+    res.json = function (params) {
+      this.params.json = params;
+      return this;
+    };
+  
+    return [req, res, next];
+  };
+  
+  const validToken = () =>
+    "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVmNzRiNTE1ZTQ4ODRkMmM1NDcwMzA3NyIsImlhdCI6MTYwMTQ4NDEyOX0.YK-fzBeDVHckGo7NzuQTRxD1_9A2qjBViZqzabGqTlk";
+  
+  it("Check auth middleware with valid token", async () => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const [req, res, next] = argForMiddleware();
+        req.get = validToken;
+        await checkAuthTokenMiddleware(req, res, next);
+        if (req.user._id) resolve();
+      } catch (err) {
+        reject(err);
+      }
+    });
+  });
+  
+  const invalidToken = () =>
+    "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVmNzRlYmQ5MGMwOGIzMzU5ODlkNzVjOCIsImlhdCI6MTYwMTY1NDAyMH0.bSaMyhIWYzjnx8aVIIDh8QwGLxY-UhmR4pjUT_nmanG";
+  
+  it("Check auth middleware with invalid token", async () => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const [req, res, next] = argForMiddleware();
+        req.get = invalidToken;
+        await checkAuthTokenMiddleware(req, res, next);
+        if (res.params.status === 401) resolve();
+      } catch (err) {
+        reject(err);
+      }
+    });
+  });
+  
+  const nonToken = () => "";
+  
+  it("Check auth middleware without token", async () => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const [req, res, next] = argForMiddleware();
+        req.get = nonToken;
+        await checkAuthTokenMiddleware(req, res, next);
+        if (res.params.status === 401) resolve();
+      } catch (err) {
+        reject(err);
+      }
+    });
+  });
+})
+
+describe('Check POST upload avatar', function () {
+  const token = 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVmN2Y0MTkxNWVmZDgxMjE0YzRlNjk5YyIsImlhdCI6MTYwMjE3NTQxOH0.9Dz4kOJ6hLT8R8gQGhNEfvSk52B_AQhajO1-q9G5W38';
+  const app = express();
+  app.use(express.urlencoded({ extended: false }))
+  app.use("/auth", authRouter);
+
+  it('Check middleware with invalid token',  (done) => {
+    request(app)
+      .get('/current')
+      .set('Authorization', token)
+      .expect(200, done)
```

Comment:
Check middleware with invalid token should be 200 ?

Reply:
no, i try 200 and 401, but always get 404, i don't understand why. i forgot remove 200